airflow/migrations/versions/c2091a80ac70_create_scheduler_state_table.py

```diff
@@ -0,0 +1,39 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""create scheduler_state table
+
+Revision ID: c2091a80ac70
+Revises: d2ae31099d61
+Create Date: 2019-11-14 16:22:10.159454
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'c2091a80ac70'
+down_revision = 'd2ae31099d61'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.create_table(
+        'scheduler_state',
+        sa.Column('state', sa.String(50), primary_key=True)
```

Comment:
If we were running multiple schedulers, this single-column, single-row isn't going to cut it. Each scheduler would need a place to report its status. To be clear: For Bluecore, this is _totally_ sufficient -- I can't imagine a situation where Bluecore would run multiple `scheduler` instances. The only reason I'm bringing this up at all is if we want to submit these changes upstream. Maybe this should be left like this for now and we should just make a ticket to support multiple schedulers later.

Reply:
Good point - I'll add a TODO here with a reference to that ticket.

Reply:
Actually...I think this would work just fine in a multi-scheduler world.  The state transition executed by the scheduler is one-way - PAUSING -> IDLE.  Each scheduler is going to check all tasks managed by all schedulers before making this transition, and each scheduler checks this single row before scheduling any jobs.  No jobs are scheduled in either the PAUSING or the IDLE states, so the only race condition here is that multiple schedulers may try to change the state from PAUSING to IDLE at the same time, but that doesn't present a problem.

Let me know if I'm missing something, but I think we're actually ok here.

Reply:
Do you care about coming back up? (`IDLE` -> `RUNNING`)?

Reply:
The scheduler does that when it boots, so it's the same one-way transition but on new schedulers.  It's done as a one-time thing at boot, not in the heartbeat loop that makes the PAUSING->IDLE transition.

Reply:
Documentation: we discussed offline and agreed that a ticket to accommodate multiple schedulers is not necessary.